extern/boostd-data/yugabyte/service.go

```diff
@@ -145,26 +162,52 @@ func (s *Store) createPieceMetadata(ctx context.Context, pieceCid cid.Cid) error
 	if err != nil {
 		return fmt.Errorf("inserting piece metadata for piece %s: %w", pieceCid, err)
 	}
+
 	return nil
 }
 
 func (s *Store) GetOffsetSize(ctx context.Context, pieceCid cid.Cid, hash mh.Multihash) (*model.OffsetSize, error) {
 	ctx, span := tracing.Tracer.Start(ctx, "store.get_offset_size")
 	defer span.End()
+	ctx, _ = tag.New(ctx, tag.Upsert(metrics.Endpoint, "yb.get_offset_size"))
+	stop := metrics.Timer(ctx, metrics.APIRequestDuration)
+	defer stop()
+	failureMetrics := true
+	defer func() {
+		if failureMetrics {
+			stats.Record(s.ctx, metrics.FailureGetOffsetSizeCount.M(1))
+		} else {
+			stats.Record(s.ctx, metrics.SuccessGetOffsetSizeCount.M(1))
+		}
+	}()
 
 	var offset, size uint64
 	qry := `SELECT BlockOffset, BlockSize FROM PieceBlockOffsetSize WHERE PieceCid = ? AND PayloadMultihash = ?`
 	err := s.session.Query(qry, pieceCid.Bytes(), hash).WithContext(ctx).Scan(&offset, &size)
 	if err != nil {
 		err = normalizePieceCidError(pieceCid, err)
+		stats.Record(s.ctx, metrics.FailureGetOffsetSizeCount.M(1))
```

Comment:
Why do we increment twice that metric -- once here and once in the defer statement?